src/hotspot/cpu/x86/vm_version_x86.hpp

```diff
@@ -300,87 +300,73 @@ class VM_Version : public Abstract_VM_Version {
+    decl(AVX512ER,          "avx512er",          30) \
+    decl(AVX512CD,          "avx512cd",          31) \
+                                                     \
+    decl(AVX512BW,          "avx512bw",          32) /* Byte and word vector instructions */ \
+    decl(AVX512VL,          "avx512vl",          33) /* EVEX instructions with smaller vector length */ \
+    decl(SHA,               "sha",               34) /* SHA instructions */ \
+    decl(FMA,               "fma",               35) /* FMA instructions */ \
+                                                     \
+    decl(VZEROUPPER,        "vzeroupper",        36) /* Vzeroupper instruction */ \
+    decl(AVX512_VPOPCNTDQ,  "avx512_vpopcntdq",  37) /* Vector popcount */ \
+    decl(AVX512_VPCLMULQDQ, "avx512_vpclmulqdq", 38) /* Vector carryless multiplication */ \
+    decl(AVX512_VAES,       "avx512_vaes",       39) /* Vector AES instruction */ \
+                                                     \
+    decl(AVX512_VNNI,       "avx512_vnni",       40) /* Vector Neural Network Instructions */ \
+    decl(FLUSH,             "clflush",           41) /* flush instruction */ \
+    decl(FLUSHOPT,          "clflushopt",        42) /* flusopth instruction */ \
+    decl(CLWB,              "clwb",              43) /* clwb instruction */ \
+                                                     \
+    decl(AVX512_VBMI2,      "avx512_vbmi2",      44) /* VBMI2 shift left double instructions */ \
+    decl(AVX512_VBMI,       "avx512_vbmi",       45) /* Vector BMI instructions */ \
+    decl(HV,                "hv",                46) /* Hypervisor instructions */
+#define DECLARE_CPU_FEATURE_FLAG(id, name, bit) CPU_##id = (1ULL << bit),
+    CPU_FEATURE_FLAGS(DECLARE_CPU_FEATURE_FLAG)
+#undef DECLARE_CPU_FEATURE_FLAG
   };
 
-#define FEATURES_NAMES \
-    "cx8",          "cmov",             "fxsr",              "ht",          \
-    "mmx",          "3dnowpref",        "sse",               "sse2",        \
-    "sse3",         "ssse3",            "sse4a",             "sse4.1",      \
-    "sse4.2",       "popcnt",           "lzcnt",             "tsc",         \
-    "tscinvbit",    "tscinv",           "avx",               "avx2",        \
-    "aes",          "erms",             "clmul",             "bmi1",        \
-    "bmi2",         "rtm",              "adx",               "avx512f",     \
-    "avx512dq",     "avx512pf",         "avx512er",          "avx512cd",    \
-    "avx512bw",     "avx512vl",         "sha",               "fma",         \
-    "vzeroupper",   "avx512_vpopcntdq", "avx512_vpclmulqdq", "avx512_vaes", \
-    "avx512_vnni",  "clflush",          "clflushopt",        "clwb",        \
-    "avx512_vmbi2", "avx512_vmbi",      "hv"
-
   static const char* _features_names[];
 
-  // NB! When adding new CPU feature detection consider updating vmStructs_x86.hpp, vmStructs_jvmci.hpp, and VM_Version::get_processor_features().
```

Comment:
No need for this comment any more as the derivative declarations are now automatically kept up to date.